internal/server/handlers.go

```diff
@@ -465,7 +466,7 @@ func (a *API) DeleteGrant(c *gin.Context, r *api.Resource) (*api.EmptyResponse,
 		}
 
 		if len(infraAdminGrants) == 1 {
-			return nil, fmt.Errorf("%w: cannot remove the last infra admin", internal.ErrForbidden)
+			return nil, fmt.Errorf("%w: cannot remove the last infra admin", internal.ErrBadRequest)
```

Comment:
This error could be a `409 Conflict`, but I think we want to save that for "duplicate entry", so bad request is probably acceptable. Otherwise we'll need to provide more context about the type of conflict error.